packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/sharedobjects/SharedObject.kt

```diff
@@ -60,10 +60,16 @@ open class SharedObject(runtimeContext: RuntimeContext? = null) {
 
   open fun onStopListeningToEvent(eventName: String) = Unit
 
+  /**
+   * Called when the shared object was released.
+   */
+  open fun wasReleased() = deallocate()
```

Comment:
Let's use the same name as on iOS: https://github.com/expo/expo/blob/c3c7fa6560365500fdb86111d672a1c1f4aa3a50/packages/expo-modules-core/ios/Core/SharedObjects/SharedObject.swift#L35-L38

```suggestion
  open fun sharedObjectDidRelease() = deallocate()
```